neo/renderer/tr_main.cpp

```diff
@@ -773,43 +755,51 @@ void R_TransformClipToDevice( const idPlane &clip, const viewDef_t *view, idVec3
 	normalized[2] = clip[2] / clip[3];
 }
 
-
 /*
 ==========================
-myGlMultMatrix
+R_MatrixMultiply
+
+Changed name to be more in line with RBDoom
 ==========================
 */
-void myGlMultMatrix( const float a[16], const float b[16], float out[16] ) {
-#if 0
-	int		i, j;
-
-	for ( i = 0 ; i < 4 ; i++ ) {
-		for ( j = 0 ; j < 4 ; j++ ) {
-			out[ i * 4 + j ] =
-				a [ i * 4 + 0 ] * b [ 0 * 4 + j ]
-				+ a [ i * 4 + 1 ] * b [ 1 * 4 + j ]
-				+ a [ i * 4 + 2 ] * b [ 2 * 4 + j ]
-				+ a [ i * 4 + 3 ] * b [ 3 * 4 + j ];
+void R_MatrixMultiply( const float a[16], const float b[16], float out[16] ) {
+	int cpuid = idLib::sys->GetProcessorId();
+
+	/* fast path use sse */
+	if ( cpuid & CPUID_SSE ) {
+		__m128 B0x = _mm_loadu_ps( &b[0] );
```

Comment:
I think this won't even compile on non-x86 (or x86_64) machines, so it must be behind some kind of `#if` or `#ifdef`

Reply:
should be fixed now havent comitted the new changes yet though (different machine so i need to setup a few things).

it does compile with mingw-w64 and clang now but i noticed i forgot to change the __SSE2__ macro to __SSE__ which caused it to fail.

might need another guard for unix / mac lets see.